acstools/acs_destripe_plus.py

```diff
@@ -495,16 +503,82 @@ def destripe_plus(inputfile, suffix='strp', stat='pmode1', maxiter=15,
         acs2d.acs2d(blctmp_name)
 
     # delete intermediate files
-    os.remove(blvtmp_name)
-    if cte_correct and os.path.isfile(blctmp_name):
-        os.remove(blctmp_name)
+    if not keep_intermediate_files:
+        os.remove(blvtmp_name)
+        if cte_correct and os.path.isfile(blctmp_name):
+            os.remove(blctmp_name)
 
     info_str = 'Done.\nFLT: {0}\n'.format(flt_name)
     if cte_correct:
         info_str += 'FLC: {0}\n'.format(flc_name)
     LOG.info(info_str)
 
 
+def crrej_plus(filelist, outroot, verbose=True):
+    """
+    Perform CRREJ and ACS2D on given BLV_TMP or BLC_TMP files.
+    The purpose of this is primarily for combining destriped
+    products from :func:`destripe_plus`.
+
+    .. note::
+
+        If you want to run this step, it is important to set
+        ``keep_intermediate_files=True`` in :func:`destripe_plus`.
+
+        This function has no TEAL nor command line interface.
+
+    Parameters
+    ----------
+    filelist : list of str
+        List of BLV_TMP or BLC_TMP files to be combined and processed.
+        Do not mix BLV and BLC in the same list, but rather run this
+        once for BLV and once for BLC separately.
+
+    outroot : str
+        Rootname for combined product, which will be named
+        ``<rootname>_crj.fits`` (for BLV inputs) or
+        ``<rootname>_crc.fits`` (for BLC inputs).
+
+    verbose : bool
+        Print informational messages. Default = True.
+
+    Raises
+    ------
+    ValueError
+        Ambiguous input list.
+
+    Examples
+    --------
+    >>> import glob
+    >>> from acstools import acs_destripe_plus
+    >>> acs_destripe_plus.destripe_plus(..., keep_intermediate_files=True)
+    >>> rootname = 'j12345678'
+    >>> blv_files = glob.glob('*_blv_tmp.fits')
+    >>> acs_destripe_plus.crrej_plus(blv_files, rootname)
+    >>> blc_files = glob.glob('*_blc_tmp.fits')
+    >>> acs_destripe_plus.crrej_plus(blc_files, rootname)
+
+    """
+    is_blv = np.all(['blv_tmp.fits' in f for f in filelist])
+    is_blc = np.all(['blc_tmp.fits' in f for f in filelist])
+
+    if is_blv is is_blc:
+        raise ValueError('Inputs must be all BLV_TMP or all BLC_TMP files')
+
+    if is_blv:
+        sfx = 'crj'
+    else:  # is_blc
+        sfx = 'crc'
+
+    tmpname = '{0}_{1}_tmp.fits'.format(outroot, sfx)
+    acsrej.acsrej(filelist, tmpname, verbose=verbose)
+    acs2d.acs2d(tmpname, verbose=verbose)  # crx_tmp -> crx
+
+    # Delete intermediate file
+    if os.path.isfile(tmpname):
```

Comment:
@jryon , do you also want an option to keep temp files here, or not? Please advise. Thanks!

Reply:
It looks like these temp files are just renamed BLV_TMP/BLC_TMP files. Since crrej_plus needs BLV_TMP/BLC_TMP files as input, these should already be available, right?

Reply:
I don't remember why I added this now, maybe to be consistent with original behavior... Should I just delete the "delete intermediate file" logic here, or you want an option to toggle it?

Reply:
Oh, I misread this before. I think an option to toggle keeping the CR-rejected BLVs around could be helpful. I like having the option to see the result of every step in the typical calibration process if desired.